File: CS50X/sentimental-readability/readability.py
# TODO

import sys


def lenghtBy(text, type):
    i = 0
    count = 0
    flag = False
    while i < len(text):
        current = text[i].upper()
        if type == 0:
            flag = current >= 'A' and current <= 'Z'
        elif type == 1:
            flag = current == ' '
        else:
            flag = is_symbol(current)

        if flag:
            count += 1

        i += 1

    # ADD COUNT FOR LAST CHARACTER = ' '
    if type == 1:
        count += 1

    return count


def is_symbol(character):
    return (character == '!' or character == '.' or character == '?')


def show_result(index):
    if index < 0:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")


def main():
    text = input("Text: ")
    letters = lenghtBy(text, 0)
    words = lenghtBy(text, 1)
    sentences = lenghtBy(text, 2)

    L = (letters / words) * 100
    S = (sentences / words) * 100

    index = 0.0588 * L - 0.296 * S - 15.8
    show_result(index)


if __name__ == "__main__":
    main()

File: CS50X/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int lenghtBy(string text, int type);
bool is_symbol(char character);
void show_result(double index);

int main(void)
{
    string text = get_string("Text: ");
    int letters = lenghtBy(text, 0);
    int words = lenghtBy(text, 1);
    int sentences = lenghtBy(text, 2);

    double L = ((double) letters / (double) words) * 100;
    double S = ((double) sentences / (double) words) * 100;

    double index = 0.0588 * L - 0.296 * S - 15.8;
    show_result(index);
}

bool is_symbol(char character)
{
    return (character == '!' || character == '.' || character == '?');
}

int lenghtBy(string text, int type)
{
    int i = 0;
    int count = 0;
    bool flag = false;
    while (i < strlen(text))
    {
        char current = toupper(text[i++]);
        switch (type)
        {
            case 0:
                flag = (current >= 'A' && current <= 'Z');
                break;
            case 1:
                flag = (current == ' ');
                break;
            default:
                flag = (is_symbol(current));
                break;
        }
        if (flag)
        {
            count++;
        }
    }
    //ADD COUNT FOR LAST CHARACTER = ' '
    if (type == 1)
    {
        count++;
    }
    return count;
}

void show_result(double index)
{
    if (index < 0)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}